pkg/tasktesting/run.go

```diff
@@ -93,33 +94,36 @@ func Run(t *testing.T, tc TestCase, testOpts TestOpts) {
 	}
 
 	taskRun, collectedLogsBuffer, err := WatchTaskRunUntilDone(t, testOpts, tr)
-	if err != nil {
-		t.Fatal(err)
+	hasSetupFailed := err != nil
+	if hasSetupFailed != tc.WantSetupFail {
+		t.Fatalf("Got: %+v, want: %+v.", hasSetupFailed, tc.WantSetupFail)
 	}
 
-	if collectedLogsBuffer.Len() > 0 {
-		testCaseContext.CollectedLogs = collectedLogsBuffer.Bytes()
-	}
+	if !hasSetupFailed {
```

Comment:
I'd probably change this section to:
```
if err != nil {
  if tc.WantSetupFail {
    return
  } else {
    t.Fatalf("Task setup failed: %s", err)
  }
}
```

Avoiding the indentation below. Those kind of early returns are pretty common in a lot of Go code.

Reply:
This was the section I felt the most uncomfortable about. Thanks for your input! 😊